### PR TITLE
extend our cosine decay to allow restart

### PR DESCRIPTION
**Pull request by vbvg2008**

### CODE REVIEW

File: fastestimator/schedule/lr_shedule.py [hunk 1]
```diff
@@ -24,19 +29,33 @@ def cosine_decay(time: int, cycle_length: int, init_lr: float, min_lr: float = 1
...
 
+    For more information, checkout SGDR: https://arxiv.org/pdf/1608.03983.pdf.
+
     Args:
         time: The current step or epoch during training starting from 1.
         cycle_length: The decay cycle length.
         init_lr: Initial learning rate to decay from.
         min_lr: Minimum learning rate.
         start: The step or epoch to start the decay schedule.
+        cycle_multiplier: The factor of which next cycle length will be multiplied.
```
Comment (TortoiseHam) [inline fastestimator/schedule/lr_shedule.py:40]: The factor by which the next cycle length will be multiplied.

File: fastestimator/schedule/lr_shedule.py [hunk 2]
```diff
@@ -24,19 +29,33 @@ def cosine_decay(time: int, cycle_length: int, init_lr: float, min_lr: float = 1
     fe.Estimator(..., traces=[s])
     ```
 
+    For more information, checkout SGDR: https://arxiv.org/pdf/1608.03983.pdf.
```
Comment (TortoiseHam) [inline fastestimator/schedule/lr_shedule.py:32]: check out

File: fastestimator/schedule/lr_shedule.py [hunk 3]
```diff
@@ -24,19 +29,33 @@ def cosine_decay(time: int, cycle_length: int, init_lr: float, min_lr: float = 1
...
+        time = time - start + 1
+        if cycle_multiplier > 1:
+            current_cycle_idx = math.ceil(
+                math.log(time * (cycle_multiplier - 1) / cycle_length + 1) / math.log(cycle_multiplier)) - 1
+            cumulative = cycle_length * (cycle_multiplier**current_cycle_idx - 1) / (cycle_multiplier - 1)
+        elif cycle_multiplier == 1:
+            current_cycle_idx = math.ceil(time / cycle_length) - 1
+            cumulative = current_cycle_idx * cycle_length
+        else:
+            raise ValueError("multiplier must be at least 1")
```
Comment (TortoiseHam) [inline fastestimator/schedule/lr_shedule.py:57]: why do we enforce that the user can't shorten the cycles?

Comment (vbvg2008) [reply fastestimator/schedule/lr_shedule.py:57]: it doesn't make sense to do that.   we want lr to become more and more stable as training progresses

File: fastestimator/schedule/lr_shedule.py (hunk 1 above)
Comment (vbvg2008) [reply fastestimator/schedule/lr_shedule.py]: y

File: fastestimator/schedule/lr_shedule.py (hunk 2 above)
Comment (vbvg2008) [reply fastestimator/schedule/lr_shedule.py]: y

**Pull request merged**